src/main/java/com/lambdaschool/foundation/models/HistoricalWeather.java

```diff
@@ -0,0 +1,106 @@
+package com.lambdaschool.foundation.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "historicalweather")
+public class HistoricalWeather extends Auditable
+{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long weatherid;
+
+    @NotNull
+    private String month;
+
+    @NotNull
+    private Double precipitation;
+
+    @NotNull
+    private Double temperature;
+
+    @ManyToOne
+    @JoinColumn(name = "cityid")
+    @NotNull
+    @JsonIgnoreProperties(value = "historicalweather")
+    private City city;
+
+    public HistoricalWeather()
+    {
+    }
+
+    public HistoricalWeather(
+        @NotNull String month,
+        @NotNull Double precipitation,
+        @NotNull Double temperature,
+        @NotNull City city)
+    {
+        this.month = month;
+        this.precipitation = precipitation;
+        this.temperature = temperature;
+        this.city = city;
+    }
+
+    public long getWeatherid()
+    {
+        return weatherid;
+    }
+
+    public void setWeatherid(long weatherid)
```

Comment:
I would recommend getting rid of this setter due to the id generation we are using.

Reply:
While we are generating the id, we still want to have the ability to change it if we want. Because the id is being generated locally on our backend, the ID isn't a parameter in any of the constructors. Being that the Id field is private we want to retain the ability to update it through the setter